### PR TITLE
Update Utf8JsonWriter for null strings, double/single formatting, and consistency.

### PR DESCRIPTION
**Pull request by ahsonkhan**
Also updated JsonReaderOptions and JsonWriterOptions to fix some nits.

cc @mairaw, @rpetrusha, @carlossanlop

### CODE REVIEW

File: xml/System.Text.Json/Utf8JsonWriter.xml [hunk 1]
```diff
@@ -56,7 +56,7 @@ To be able to format the output with indentation and white space OR to skip vali
       <Docs>
         <param name="bufferWriter">The destination for writing JSON text.</param>
         <param name="options">Defines the customized behavior of the <see cref="T:System.Text.Json.Utf8JsonWriter" /> By default, it writes minimized JSON (with no extra whitespace) and validates that the JSON being written is structurally valid according to the JSON RFC.</param>
-        <summary>Constructs a new <see cref="T:System.Text.Json.Utf8JsonWriter" /> instance with a specified <see langword="System.Buffers.IBufferWriter&lt;System.Byte&gt;" />.</summary>
```
Comment (ahsonkhan) [inline xml/System.Text.Json/Utf8JsonWriter.xml]: I don't know why langword was chosen for this.

File: xml/System.Text.Json/Utf8JsonWriter.xml [hunk 2]
```diff
@@ -322,8 +322,8 @@ In the case of <xref:System.IO.Stream>, this writes the data to the stream and f
         <ReturnType>System.Text.Json.JsonWriterOptions</ReturnType>
       </ReturnValue>
       <Docs>
-        <summary>Gets the custom behavior when writing JSON using this instance, which indicates whether to format the output while writing and whether to skip structural JSON validation.</summary>
-        <value>An object that defines the behavior of this instance for formatting and validation.</value>
+        <summary>Gets the custom behavior when writing JSON using this instance, which indicates whether to format the output while writing, whether to skip structural JSON validation, and which characters to escape.</summary>
+        <value>Defines the behavior of this instance of the writer for formatting, validation, and escaping.</value>
```
Comment (ahsonkhan) [inline xml/System.Text.Json/Utf8JsonWriter.xml]: JsonWriterOptions is a struct, not a class.

File: xml/System.Text.Json/JsonReaderOptions.xml [hunk 3]
```diff
@@ -116,7 +116,7 @@ By default, the reader throws a <xref:System.Text.Json.JsonException> if it enco
 
 ## Remarks
 
-Reading past this depth will throw a <exception cref="T:System.Text.Json.JsonException>.
+Reading past this depth will throw a <xref:System.Text.Json.JsonException>.
```
Comment (ahsonkhan) [inline xml/System.Text.Json/JsonReaderOptions.xml]: This was not showing up correctly:
https://docs.microsoft.com/en-us/dotnet/api/system.text.json.jsonreaderoptions.maxdepth?view=netcore-3.0

image

File: xml/System.Text.Json/Utf8JsonWriter.xml [hunk 4]
```diff
@@ -806,7 +806,7 @@ The property name should already be escaped when the instance of <xref:System.Te
         <Parameter Name="utf8Value" Type="System.ReadOnlySpan&lt;System.Byte&gt;" />
       </Parameters>
       <Docs>
-        <param name="utf8Value">The UTF-8 encoded value to be written as a JSON comment within /*..*/.</param>
+        <param name="utf8Value">The UTF-8 encoded value to be written as a JSON comment within `/*..*/`.</param>
```
Comment (ahsonkhan) [inline xml/System.Text.Json/Utf8JsonWriter.xml]: This was not showing up correctly.

https://docs.microsoft.com/en-us/dotnet/api/system.text.json.utf8jsonwriter.writecommentvalue?view=netcore-3.0

image

File: xml/System.Text.Json/Utf8JsonWriter.xml [hunk 5]
```diff
@@ -2756,7 +2756,7 @@ The property name should already be escaped when the instance of <xref:System.Te
         </remarks>
         <exception cref="T:System.InvalidOperationException">The depth of the JSON has exceeded the maximum depth of 1000.
     
-- or -
+-or-
```
Comment (ahsonkhan) [inline xml/System.Text.Json/Utf8JsonWriter.xml]: This was not showing up correctly:
https://docs.microsoft.com/en-us/dotnet/api/system.text.json.utf8jsonwriter.writestartarray?view=netcore-3.0#System_Text_Json_Utf8JsonWriter_WriteStartArray_System_Text_Json_JsonEncodedText_

image

File: xml/System.Text.Json/Utf8JsonWriter.xml [hunk 6]
```diff
@@ -3155,6 +3155,8 @@ The property name and value are escaped before writing.
 
 The property name and value are escaped before writing.
 
+If <paramref name="value"/> is <see langword="null"/> the JSON null value is written, as if <see cref="WriteNull(System.ReadOnlySpan{byte})"/> was called.
```
Comment (ahsonkhan) [inline xml/System.Text.Json/Utf8JsonWriter.xml]: Addressing recent changes around "null:
https://github.com/dotnet/dotnet-api-docs/pull/3081#discussion_r317874872

> This PR updated the description around nulls in a few places. We should make sure to capture that:
> [dotnet/corefx#39344](https://github.com/dotnet/corefx/pull/39344)

File: xml/System.Text.Json/Utf8JsonWriter.xml [hunk 7]
```diff
@@ -56,7 +56,7 @@ To be able to format the output with indentation and white space OR to skip vali
       <Docs>
         <param name="bufferWriter">The destination for writing JSON text.</param>
         <param name="options">Defines the customized behavior of the <see cref="T:System.Text.Json.Utf8JsonWriter" /> By default, it writes minimized JSON (with no extra whitespace) and validates that the JSON being written is structurally valid according to the JSON RFC.</param>
-        <summary>Constructs a new <see cref="T:System.Text.Json.Utf8JsonWriter" /> instance with a specified <see langword="System.Buffers.IBufferWriter&lt;System.Byte&gt;" />.</summary>
+        <summary>Constructs a new <see cref="T:System.Text.Json.Utf8JsonWriter" /> instance with a specified <see cref="T:System.Buffers.IBufferWriter`1" /> to write the output to.</summary>
```
Comment (ahsonkhan) [inline xml/System.Text.Json/Utf8JsonWriter.xml]: Looks like the build didn't like using 
```text
`IBufferWriter<byte>`
```

So, I decided to use the generic one. Is there a correct way to specify that the T is byte?

File: xml/System.Text.Json/JsonReaderOptions.xml (hunk 3 above)
Comment (carlossanlop) [reply xml/System.Text.Json/JsonReaderOptions.xml]: Yikes, thanks for catching that. They should've been caught with a failing build.

File: xml/System.Text.Json/Utf8JsonWriter.xml (hunk 1 above)
Comment (carlossanlop) [reply xml/System.Text.Json/Utf8JsonWriter.xml]: I think @rpetrusha [did this as a workaround](https://github.com/dotnet/dotnet-api-docs/pull/2576) to correctly render a link with a specific T. It won't show you a link, it will show up like `true` or `false`, but at least it will render fine.

Your other options are what you did (use `1) or do what I did with Intrinsics, add a link but explain the expected T separately (explained in your other PR).

File: xml/System.Text.Json/Utf8JsonWriter.xml (hunk 7 above)
Comment (carlossanlop) [reply xml/System.Text.Json/Utf8JsonWriter.xml]: See my comment above.

File: xml/System.Text.Json/Utf8JsonWriter.xml [hunk 8]
```diff
@@ -80,7 +80,7 @@ To be able to format the output with indentation and white space OR to skip vali
       <Docs>
         <param name="utf8Json">The destination for writing JSON text.</param>
         <param name="options">Defines the customized behavior of the <see cref="T:System.Text.Json.Utf8JsonWriter" /> By default, it writes minimized JSON (with no extra whitespace) and validates that the JSON being written is structurally valid according to the JSON RFC.</param>
-        <summary>Constructs a new <see cref="T:System.Text.Json.Utf8JsonWriter" /> instance with a specified <paramref name="utf8Json" />.</summary>
+        <summary>Constructs a new <see cref="T:System.Text.Json.Utf8JsonWriter" /> instance with a specified <see cref="T:System.IO.Stream" /> to write the output to.</summary>
```
Comment (carlossanlop) [inline xml/System.Text.Json/Utf8JsonWriter.xml]: We normally mention streams with plain language:

```suggestion
        <summary>Constructs a new <see cref="T:System.Text.Json.Utf8JsonWriter" /> instance with a specified stream to write the output to.</summary>
```

File: xml/System.Text.Json/Utf8JsonWriter.xml (hunk 2 above)
Comment (carlossanlop) [reply xml/System.Text.Json/Utf8JsonWriter.xml]: It's my understanding that `<value>` needs to explain the data type and what it represents. So you could prepend it with "struct instance" for example. Also, as I mentioned in your other PR, in our official docs for "C# objects", it seems that we use "object" to describe a struct instance too.

File: xml/System.Text.Json/JsonReaderOptions.xml (hunk 3 above)
Comment (mairaw) [reply xml/System.Text.Json/JsonReaderOptions.xml]: There are a bunch of links that the current build system doesn't report as broken. This should improve once we move to SDP. But we'd have to test this scenario when we migrate. It might be that doesn't validate anything because it's expecting Markdown so it just renders the text.

File: xml/System.Text.Json/Utf8JsonWriter.xml (hunk 7 above)
Comment (mairaw) [inline xml/System.Text.Json/Utf8JsonWriter.xml]: Somehow GitHub won't let me do a suggestion for this line but it should be something like:

`Initializes a new instance of the <see cref="T:System.Text.Json.Utf8JsonWriter" /> class using the specified buffer and customization options.`

File: xml/System.Text.Json/Utf8JsonWriter.xml (hunk 8 above)
Comment (mairaw) [reply xml/System.Text.Json/Utf8JsonWriter.xml]: I'd add to Carlos suggestion and use the text from our wiki - https://github.com/dotnet/dotnet-api-docs/wiki/Summary%3A-Constructor

```suggestion
        <summary>Initializes a new instance of the <see cref="T:System.Text.Json.Utf8JsonWriter" /> class using the specified stream to write the output to and customization options.</summary>
```

File: xml/System.Text.Json/Utf8JsonWriter.xml [hunk 9]
```diff
@@ -80,7 +80,7 @@ To be able to format the output with indentation and white space OR to skip vali
       <Docs>
         <param name="utf8Json">The destination for writing JSON text.</param>
         <param name="options">Defines the customized behavior of the <see cref="T:System.Text.Json.Utf8JsonWriter" /> By default, it writes minimized JSON (with no extra whitespace) and validates that the JSON being written is structurally valid according to the JSON RFC.</param>
```
Comment (mairaw) [inline xml/System.Text.Json/Utf8JsonWriter.xml]:
```suggestion
        <param name="options">One of the enumeration values that that defines the customized behavior of the <see cref="T:System.Text.Json.Utf8JsonWriter" /> By default, it writes minimized JSON (with no extra white space) and validates that the JSON being written is structurally valid according to the JSON RFC.</param>
```

File: xml/System.Text.Json/Utf8JsonWriter.xml [hunk 10]
```diff
@@ -56,7 +56,7 @@ To be able to format the output with indentation and white space OR to skip vali
       <Docs>
         <param name="bufferWriter">The destination for writing JSON text.</param>
         <param name="options">Defines the customized behavior of the <see cref="T:System.Text.Json.Utf8JsonWriter" /> By default, it writes minimized JSON (with no extra whitespace) and validates that the JSON being written is structurally valid according to the JSON RFC.</param>
```
Comment (mairaw) [inline xml/System.Text.Json/Utf8JsonWriter.xml]:
```suggestion
        <param name="options">One of the enumeration values that defines the customized behavior of the <see cref="T:System.Text.Json.Utf8JsonWriter" /> By default, it writes minimized JSON (with no extra white space) and validates that the JSON being written is structurally valid according to the JSON RFC.</param>
```

File: xml/System.Text.Json/Utf8JsonWriter.xml (hunk 2 above)
Comment (mairaw) [reply xml/System.Text.Json/Utf8JsonWriter.xml]: Maybe something like this? It shouldn't start with a verb here. 

```suggestion
        <value>The custom behavior of this instance of the writer for formatting, validating, and escaping.</value>
```

File: xml/System.Text.Json/Utf8JsonWriter.xml (hunk 4 above)
Comment (mairaw) [reply xml/System.Text.Json/Utf8JsonWriter.xml]: Once we merge this, we should also test this with our markdig branch since rendering could change as well.
Ideally we should try to use valid ecmaxml here but not sure how it would render:

```suggestion
        <param name="utf8Value">The UTF-8 encoded value to be written as a JSON comment within <c>/*..*/<\c>.</param>
```

If this works, we should apply the same fix to other places in this doc. Otherwise, if we keep backticks, we should test if this would render in the output IntelliSense.

File: xml/System.Text.Json/Utf8JsonWriter.xml [hunk 11]
```diff
@@ -1167,7 +1167,7 @@ The property name is escaped before writing.
 
 ## Remarks
 
-Writes the <xref:System.Double> using the default <xref:System.Buffers.StandardFormat> (that is, 'G').
+Writes the <xref:System.Double> using the default <xref:System.Buffers.StandardFormat> (that is, 'G') on .NET Core 3.0 or higher. Uses 'G17' on any other framework.
```
Comment (mairaw) [inline xml/System.Text.Json/Utf8JsonWriter.xml]:
```suggestion
Writes the <xref:System.Double> using the default <xref:System.Buffers.StandardFormat> (that is, 'G') on .NET Core 3.0 or later versions. Uses 'G17' on any other framework.
```

File: xml/System.Text.Json/Utf8JsonWriter.xml (hunk 6 above)
Comment (mairaw) [reply xml/System.Text.Json/Utf8JsonWriter.xml]:
```suggestion
If `value` is `null`, the JSON null value is written as if the <xref:System.Text.Json.Utf8JsonWriter.WriteNull(System.ReadOnlySpan{System.Byte})"> method was called.
```

File: xml/System.Text.Json/Utf8JsonWriter.xml [hunk 12]
```diff
@@ -3415,6 +3417,8 @@ The property name and value are escaped before writing.
 
 The property name and value are escaped before writing.
 
+If <paramref name="value"/> is <see langword="null"/> the JSON null value is written, as if <see cref="WriteNull(System.ReadOnlySpan{char})"/> was called.
```
Comment (mairaw) [inline xml/System.Text.Json/Utf8JsonWriter.xml]:
```suggestion
If `value` is `null`, the JSON null value is written as if the <xref:System.Text.Json.Utf8JsonWriter.WriteNull(System.ReadOnlySpan{System.Char})"> method was called.
```

File: xml/System.Text.Json/Utf8JsonWriter.xml [hunk 13]
```diff
@@ -3680,6 +3684,8 @@ The property name and value are escaped before writing.
 
 The property name and value are escaped before writing.
 
+If <paramref name="value"/> is <see langword="null"/> the JSON null value is written, as if <see cref="WriteNull(string)"/> were called.
```
Comment (mairaw) [inline xml/System.Text.Json/Utf8JsonWriter.xml]:
```suggestion
If `value` is `null`, the JSON null value is written as if the <xref:System.Text.Json.Utf8JsonWriter.WriteNull(System.String)"> method was called.
```

File: xml/System.Text.Json/Utf8JsonWriter.xml [hunk 14]
```diff
@@ -3909,6 +3915,8 @@ The property name should already be escaped when the instance of <xref:System.Te
 
 The value is escaped before writing.
 
+If <paramref name="value"/> is <see langword="null"/> the JSON null value is written, as if <see cref="WriteNull(System.Text.Json.JsonEncodedText)"/> was called.
```
Comment (mairaw) [inline xml/System.Text.Json/Utf8JsonWriter.xml]:
```suggestion
If `value` is `null`, the JSON null value is written as if the <xref:System.Text.Json.Utf8JsonWriter.WriteNull(System.Text.Json.JsonEncodedText)"> method was called.
```

File: xml/System.Text.Json/Utf8JsonWriter.xml [hunk 15]
```diff
@@ -3947,6 +3955,8 @@ The property name should already be escaped when the instance of <xref:System.Te
...
```
Comment (mairaw) [inline xml/System.Text.Json/Utf8JsonWriter.xml]:
```suggestion
If `value` is `null`, the JSON null value is written as if the <xref:System.Text.Json.Utf8JsonWriter.WriteNull(System.Text.Json.JsonEncodedText)"> method was called.
```

File: xml/System.Text.Json/Utf8JsonWriter.xml [hunk 16]
```diff
@@ -4185,6 +4195,8 @@ The value is escaped before writing.
 
 The value is escaped before writing.
 
+If <paramref name="value"/> is <see langword="null"/> the JSON null value is written, as if <see cref="WriteNullValue"/> was called.
```
Comment (mairaw) [inline xml/System.Text.Json/Utf8JsonWriter.xml]:
```suggestion
If `value` is `null`, the JSON null value is written as if the <xref:System.Text.Json.Utf8JsonWriter.WriteNullValue"> method was called.
```

File: xml/System.Text.Json/Utf8JsonWriter.xml (hunk 6 above)
Comment (ahsonkhan) [reply xml/System.Text.Json/Utf8JsonWriter.xml]: Why not use the langword null here (rather than back-ticks)?

Comment (mairaw) [reply xml/System.Text.Json/Utf8JsonWriter.xml]: You're in Markdown land here in the remarks. 

This is produced if you use that:
image

File: xml/System.Text.Json/Utf8JsonWriter.xml (hunk 7 above)
Comment (carlossanlop) [reply xml/System.Text.Json/Utf8JsonWriter.xml]: It happens to me to sometimes, @mairaw. Press Control+G on the comment textbox, or manually add the suggestion text. Here's your suggestion:
```suggestion
        <summary>Initializes a new instance of the <see cref="T:System.Text.Json.Utf8JsonWriter" /> class using the specified buffer and customization options.</summary>
```

File: xml/System.Text.Json/Utf8JsonWriter.xml [hunk 17]
```diff
@@ -821,7 +821,7 @@ The comment value is not escaped before writing.
         
 -or-
 
-<paramref name="utf8Value" /> contains a comment delimiter (that is, */).</exception>
+<paramref name="utf8Value" /> contains a comment delimiter (that is, `*/`).</exception>
```
Comment (carlossanlop) [inline xml/System.Text.Json/Utf8JsonWriter.xml]:
```suggestion
<paramref name="utf8Value" /> contains a comment delimiter (that is, <c>*/</c>).</exception>
```

File: xml/System.Text.Json/Utf8JsonWriter.xml [hunk 18]
```diff
@@ -843,7 +843,7 @@ The comment value is not escaped before writing.
         <Parameter Name="value" Type="System.ReadOnlySpan&lt;System.Char&gt;" />
       </Parameters>
       <Docs>
-        <param name="value">The UTF-16 encoded value to be written as a UTF-8 transcoded JSON comment within /*..*/.</param>
+        <param name="value">The UTF-16 encoded value to be written as a UTF-8 transcoded JSON comment within `/*..*/`.</param>
```
Comment (carlossanlop) [inline xml/System.Text.Json/Utf8JsonWriter.xml]:
```suggestion
        <param name="value">The UTF-16 encoded value to be written as a UTF-8 transcoded JSON comment within <c>/*..*/</c>.</param>
```

File: xml/System.Text.Json/Utf8JsonWriter.xml [hunk 19]
```diff
@@ -854,11 +854,11 @@ The comment value is not escaped before writing.
 
           ]]></format>
         </remarks>
-        <exception cref="T:System.ArgumentException">The specified value is too large OR.
+        <exception cref="T:System.ArgumentException">The specified value is too large.
         
 -or-
 
-<paramref name="value" /> contains a comment delimiter (that is, */).</exception>
+<paramref name="value" /> contains a comment delimiter (that is, `*/`).</exception>
```
Comment (carlossanlop) [inline xml/System.Text.Json/Utf8JsonWriter.xml]:
```suggestion
<paramref name="value" /> contains a comment delimiter (that is, <c>*/</c>).</exception>
```

File: xml/System.Text.Json/Utf8JsonWriter.xml [hunk 20]
```diff
@@ -880,7 +880,7 @@ The comment value is not escaped before writing.
         <Parameter Name="value" Type="System.String" />
       </Parameters>
       <Docs>
-        <param name="value">The UTF-16 encoded value to be written as a UTF-8 transcoded JSON comment within /*..*/.</param>
+        <param name="value">The UTF-16 encoded value to be written as a UTF-8 transcoded JSON comment within `/*..*/`.</param>
```
Comment (carlossanlop) [inline xml/System.Text.Json/Utf8JsonWriter.xml]:
```suggestion
        <param name="value">The UTF-16 encoded value to be written as a UTF-8 transcoded JSON comment within <c>/*..*/</c>.</param>
```

File: xml/System.Text.Json/Utf8JsonWriter.xml [hunk 21]
```diff
@@ -895,7 +895,7 @@ The comment value is not escaped before writing.
         
 -or-
 
-<paramref name="value" /> contains a comment delimiter (that is, */).</exception>
+<paramref name="value" /> contains a comment delimiter (that is, `*/`).</exception>
```
Comment (carlossanlop) [inline xml/System.Text.Json/Utf8JsonWriter.xml]:
```suggestion
<paramref name="value" /> contains a comment delimiter (that is, <c>*/</c>).</exception>
```

File: xml/System.Text.Json/Utf8JsonWriter.xml (hunk 10 above)
Comment (Jozkee) [reply xml/System.Text.Json/Utf8JsonWriter.xml]: Adding missing dot to @mairaw's suggestion.
```suggestion
        <param name="options">One of the enumeration values that defines the customized behavior of the <see cref="T:System.Text.Json.Utf8JsonWriter" />. By default, it writes minimized JSON (with no extra white space) and validates that the JSON being written is structurally valid according to the JSON RFC.</param>

```

File: xml/System.Text.Json/Utf8JsonWriter.xml (hunk 9 above)
Comment (Jozkee) [reply xml/System.Text.Json/Utf8JsonWriter.xml]:
```suggestion
        <param name="options">One of the enumeration values that that defines the customized behavior of the <see cref="T:System.Text.Json.Utf8JsonWriter" />. By default, it writes minimized JSON (with no extra white space) and validates that the JSON being written is structurally valid according to the JSON RFC.</param>

```

File: xml/System.Text.Json/Utf8JsonWriter.xml [hunk 22]
```diff
@@ -383,7 +383,7 @@ The <xref:System.Text.Json.Utf8JsonWriter> will continue to use the original wri
 
 ## Remarks
 
-The <xref:System.Text.Json.Utf8JsonWriter> will continue to use the original writer options but now writes to `bufferWriter` as the new destination.
+The <xref:System.Text.Json.Utf8JsonWriter> will continue to use the original writer options but now write to `bufferWriter` as the new destination.
```
Comment (Jozkee) [inline xml/System.Text.Json/Utf8JsonWriter.xml]: Why did you remove the singular "s"? Utf8JsonWriter is still a singular.

File: xml/System.Text.Json/Utf8JsonWriter.xml (hunk 10 above)
Comment (ahsonkhan) [reply xml/System.Text.Json/Utf8JsonWriter.xml]: The options isn't an enum though.

File: xml/System.Text.Json/Utf8JsonWriter.xml (hunk 4 above)
Comment (ahsonkhan) [reply xml/System.Text.Json/Utf8JsonWriter.xml]: Shows up fine atm, so leaving it as is for now:
image

File: xml/System.Text.Json/Utf8JsonWriter.xml (hunk 6 above)
Comment (ahsonkhan) [reply xml/System.Text.Json/Utf8JsonWriter.xml]: Ah I see. Yep, that needs to be fixed. Thanks, @mairaw.

File: xml/System.Text.Json/Utf8JsonWriter.xml (hunk 7 above)
Comment (ahsonkhan) [reply xml/System.Text.Json/Utf8JsonWriter.xml]: IBufferWriter isn't a buffer though. I would prefer to keep that as is, and also specify that the IBW is where we are "writing the output to".

Any better way to word it to keep both those things?

File: xml/System.Text.Json/Utf8JsonWriter.xml (hunk 22 above)
Comment (ahsonkhan) [reply xml/System.Text.Json/Utf8JsonWriter.xml]: Seemed to make more grammatical sense to my ear.

**Comment by ahsonkhan** [comment]
Anything left to do or is this good to merge?

File: xml/System.Text.Json/Utf8JsonWriter.xml (hunk 22 above)
Comment (mairaw) [reply xml/System.Text.Json/Utf8JsonWriter.xml]: I'm on the fence on this one. @rpetrusha can you help?

File: xml/System.Text.Json/Utf8JsonWriter.xml (hunk 4 above)
Comment (mairaw) [reply xml/System.Text.Json/Utf8JsonWriter.xml]: Yes, I saw that. Not sure how it would render in IntelliSense though. Hopefully, it will be stripped out. I can create a PR after this one is merged to test that,

File: xml/System.Text.Json/Utf8JsonWriter.xml [hunk 23]
```diff
@@ -2756,7 +2756,7 @@ The property name should already be escaped when the instance of <xref:System.Te
         </remarks>
         <exception cref="T:System.InvalidOperationException">The depth of the JSON has exceeded the maximum depth of 1000.
```
Comment (mairaw) [inline xml/System.Text.Json/Utf8JsonWriter.xml]: nit: I could also do this in a separate PR

```suggestion
        <exception cref="T:System.InvalidOperationException">The depth of the JSON has exceeded the maximum depth of 1,000.
```

**Comment by mairaw** [comment]
I'll merge this and fix the grammar if needed on a separate PR.

File: xml/System.Text.Json/Utf8JsonWriter.xml (hunk 22 above)
Comment (carlossanlop) [reply xml/System.Text.Json/Utf8JsonWriter.xml]: I agree with @Jozkee, The Utf8JsonWriter is who's writing. It writes.

Comment (mairaw) [reply xml/System.Text.Json/Utf8JsonWriter.xml]: The other option would be to add will write

Comment (rpetrusha) [reply xml/System.Text.Json/Utf8JsonWriter.xml]: The tense of write is future, so it should be "write" rather than "writes". As Maira points out, it could also be changed to "will write", and that's fine, but it's not necessary.
